ui/js/reactrevisions.js

```diff
@@ -0,0 +1,123 @@
+            )
+        );
+    }
+});
+var pushlogRevisionComponentFactory = React.createFactory(pushlogRevisionComponent);
+
+// via revisionsClone.html
+var revisionItemComponent = React.createClass({
+    displayName: 'revisionItemComponent',
+    propTypes: {
+        revision: React.PropTypes.object.isRequired,
+        repo: React.PropTypes.object.isRequired,
+        linkifyBugsFilter: React.PropTypes.func.isRequired,
+        initialsFilter: React.PropTypes.func.isRequired
+    },
+    render: function() {
+        var email, name, userTokens, escapedComment, escapedCommentHTML, initialsHTML, tags;
+
+        userTokens = this.props.revision.author.split(/[<>]+/);
+        name = userTokens[0];
+        if (userTokens.length > 1) email = userTokens[1];
+        initialsHTML = { __html: this.props.initialsFilter(name) };
+
+        escapedComment = _.escape(this.props.revision.comments.split('\n')[0]);
+        escapedCommentHTML = { __html: this.props.linkifyBugsFilter(escapedComment) };
+
+        tags = "";
+        if (escapedComment.search("Backed out") >= 0 ||
+            escapedComment.search("Back out") >= 0) {
+            tags += "backout ";
+        }
+        tags = tags.trim();
+
+        return React.DOM.li(
+            { className: 'clearfix' },
+            React.DOM.span({
+                className: 'revision',
+                'data-tags': tags
+            },
+            React.DOM.span(
+                { className: 'revision-holder' },
+                React.DOM.a({
+                    title: `Open revision ${this.props.revision.revision} on ${this.props.repo.url}`,
+                    href: this.props.repo.getRevisionHref(this.props.revision.revision),
+                    'data-ignore-job-clear-on-click': true
+                },
+                    this.props.revision.revision.substring(0, 12)
+                )),
+                React.DOM.span({
+                    title: `${name}: ${email}`,
+                    dangerouslySetInnerHTML: initialsHTML
+                }),
+                React.DOM.span(
+                    { title: escapedComment },
+                    React.DOM.span(
+                        { className: 'revision-comment' },
+                        React.DOM.em({ dangerouslySetInnerHTML: escapedCommentHTML })
+                    )
+                )
+            )
+        );
+    }
+});
+var revisionItemComponentFactory = React.createFactory(revisionItemComponent);
+
+var revisionListComponent = React.createClass({
+    displayName: 'revisionListComponent',
+    propTypes: {
+        resultset: React.PropTypes.object.isRequired,
+        repo: React.PropTypes.object.isRequired,
+        $injector: React.PropTypes.object.isRequired
+    },
+    render: function() {
+        var repo = this.props.repo;
+        var pushlogEl = null;
+
+        if (this.props.resultset.revision_count > this.props.resultset.revisions.length) {
```

Comment:
You might add a comment here just saying this is creating the "and more..." thing in cease it's needed.